ethereum/check_swap_factory_contract_test.go

```diff
@@ -93,18 +102,36 @@ func TestCheckSwapFactoryContractCode(t *testing.T) {
 
 // Tests that we fail when the wrong contract byte code is found
 func TestCheckSwapFactoryContractCode_fail(t *testing.T) {
-	ec, chainID := tests.NewEthClient(t)
+	ec, _ := tests.NewEthClient(t)
 	pk := tests.GetMakerTestKey(t)
-	txOpts, err := bind.NewKeyedTransactorWithChainID(pk, chainID)
-	require.NoError(t, err)
 
-	// Deploying an arbitrary contract that won't match the swap factory contract
-	contractAddr, tx, _, err :=
-		DeployERC20Mock(txOpts, ec, "ERC20Mock", "MOCK", ethcommon.Address{0x1}, big.NewInt(9999))
-	require.NoError(t, err)
-	_, err = block.WaitForReceipt(context.Background(), ec, tx.Hash())
+	// Deploy a forwarder contract and then try to verify it as SwapFactory contract
+	contractAddr := deployForwarder(t, ec, pk)
+	_, err := CheckSwapFactoryContractCode(context.Background(), ec, contractAddr)
+	require.ErrorIs(t, err, errInvalidSwapContract)
+}
+
+func TestGoerliContract(t *testing.T) {
+	// comment out the next line to test the default goerli contract
+	t.Skip("requires access to non-vetted external goerli node")
+	const goerliEndpoint = "https://ethereum-goerli-rpc.allthatnode.com"
+	// you can temporarily put a funded goerli key below to deploy the test contract
+	const goerliKey = ""
+	ec, err := ethclient.Dial(goerliEndpoint)
 	require.NoError(t, err)
+	defer ec.Close()
 
-	_, err = CheckSwapFactoryContractCode(context.Background(), ec, contractAddr)
-	require.ErrorIs(t, err, errInvalidSwapContract)
+	parsedTFAddr, err := CheckSwapFactoryContractCode(context.Background(), ec, common.StagenetConfig.ContractAddress)
+	if errors.Is(err, errInvalidSwapContract) && goerliKey != "" {
```

Comment:
should there be an `errors.Is(err, errInvalidSwapContract) && goerliKey == ""` error case?

Reply:
We fail this branch of the `if` on line 132. The alternate branch fails on 134 if the error is not nil.